src/artemis/zocalo_interaction.py

```diff
@@ -49,3 +62,47 @@ def run_end(data_collection_id: int):
             "ispyb_dcid": data_collection_id,
         }
     )
+
+
+def wait_for_result(data_collection_id: int, timeout: int = TIMEOUT) -> Point3D:
+    """Block until a result is recieved from Zocalo.
+    Args:
+        data_collection_id (int): The ID of the data collection representing the gridscan in ISpyB
+        timeout (float): The time in seconds to wait for the result to be recieved.
+    Returns:
+        Point in grid co-ordinates that is the centre point to move to
+    """
+    transport = _get_zocalo_connection()
+    result_recieved = None
+
+    def receive_result(
+        rw: workflows.recipe.RecipeWrapper, header: dict, message: dict
+    ) -> None:
+        print(f"Received {message}")
+        recipe_parameters = rw.recipe_step["parameters"]
+        print(f"Recipe step parameters: {recipe_parameters}")
+        if recipe_parameters["dcid"] == str(data_collection_id):
+            transport.ack(header)
+            nonlocal result_recieved
+            result_recieved = Point3D(*message["max_voxel"])
+        else:
+            print(
+                f"Warning: results for {recipe_parameters['dcid']} recieved but expected {data_collection_id}"
+            )
```

Comment:
With `acknowledgment=True` you need to explicitly `ack` or `nack` each message otherwise you won't receive any subsequent messages until you have done so:
```suggestion
            )
            transport.ack(header)
```